saleor/graphql/checkout/mutations/checkout_lines_delete.py

```diff
@@ -47,16 +48,20 @@ class Meta:
         ]
 
     @classmethod
-    def validate_lines(cls, checkout, lines_to_delete):
+    def validate_lines(cls, checkout: Checkout, lines_to_delete_ids: list[str]):
         lines = checkout.lines.all()
         all_lines_ids = [str(line.id) for line in lines]
-        invalid_line_ids = list()
-        for line_to_delete in lines_to_delete:
-            if line_to_delete not in all_lines_ids:
-                line_to_delete = graphene.Node.to_global_id(
-                    "CheckoutLine", line_to_delete
+        gift_line_ids = [str(line.id) for line in lines if line.is_gift]
+        invalid_line_ids, gift_line_to_delete_ids = list(), list()
+        for line_to_delete_id in lines_to_delete_ids:
```

Comment:
You could use `set` instead of loop. 
```python
invalid_line_ids = set(lines_to_delete_ids).difference(all_lines_ids)
gift_line_to_delete_ids = set(gift_line_ids).intersection(lines_to_delete_ids)
```

Reply:
Nice, thanks